src/main/java/org/apache/mesos/scheduler/plan/Element.java

```diff
@@ -0,0 +1,112 @@
+package org.apache.mesos.scheduler.plan;
+
+import org.apache.mesos.Protos.TaskStatus;
+import org.apache.mesos.scheduler.Observable;
+import org.apache.mesos.scheduler.plan.strategy.Strategy;
+
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * An Element is a generic component in a tree of elements.  It is the basic unit for the description of Plans used to
+ * launch Mesos Tasks. {@link Plan}s, @{link Phase}s and {@link Block}s are all implementations of Elements.  Elements
+ * have Strategies (see: {@link Strategy} associated with them which describe when child elements should be presented as
+ * candidates for execution.
+ *
+ * @param <C> is the type of child {@link Element}s.
+ */
+public interface Element<C extends Element> extends Observable {
```

Comment:
Would it make sense to have a base `Element` class without anything to do with child elements (`Block` would be one of these), and then a `ParentElement extends Element` class that has the child element templating and `getChildren()`/`getStrategy()`?